pkg/storageops/gce/gce.go

```diff
@@ -651,6 +677,46 @@ func (s *gceOps) getDisksFromAllZones(labels map[string]string) (map[string]*com
 	return response, nil
 }
 
+func (s *gceOps) diskIDToBlockDevPathWithRetry(devPath string) (string, error) {
+	var (
+		retryCount int
+		path       string
+		err        error
+	)
+
+	for {
+		if path, err = s.diskIDToBlockDevPath(devPath); err == nil {
+			return path, nil
+		}
+		logrus.Warnf(err.Error())
+		retryCount++
+		if retryCount >= devicePathMaxRetryCount {
+			break
+		}
+		time.Sleep(devicePathRetryInterval)
+	}
+	return "", err
+}
+
+func (s *gceOps) diskIDToBlockDevPath(devPath string) (string, error) {
```

Comment:
@harsh-px If this function is called only for storageops perspective, should it return an error if it is unable to find a symbolic link to an actual device path?

Reply:
Yes, this function is expected to pass for all paths (symbolic and regular). So any error would be unexpected and not ignored.

If storageops fails to find the dev path and does not return error, the path will not be useful to callers who expected a block device path.